### PR TITLE
revcomp-begin edits

### PR DESCRIPTION
**Pull request by psahabu**
@bharsh:
- converted ‘data’ from a byte array to a string
- simplified stdin/stdout usage
- simplified logic
- cleaned up syntax

### CODE REVIEW

File: test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl [hunk 1]
```diff
@@ -1,73 +1,69 @@
...
-  const fileLen = inFile.length();
-  var data : [1..fileLen] uint(8);
-  var r = inFile.reader(locking=false);
-
+proc main() {
   // initialize complement table
   for i in 0..#pairs.length by 2 {
-    table[pairs.buff[i]] = pairs.buff[i+1];      // uppercase
+    table[pairs.buff[i]] = pairs.buff[i+1];        // uppercase
     if pairs.buff[i] != newLineChar then
```
Comment (benharsh) [inline test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl]: Though less performant, I wonder if we should do this instead for elegance:
```chpl
if pairs[i] != "\n" then
```

File: test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl [hunk 2]
```diff
@@ -1,73 +1,69 @@
...
-  var numRead  : int;
+  
+  var data : string; 
   var idx = 1;
   var start = 0;
 
   // sync statements wait for all tasks inside them to complete
   sync {
-    while r.readline(data, numRead, idx) {
+    var data_in : string;
```
Comment (benharsh) [inline test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl]: Can we call this something other than "data_in"? I think it's a little too similar to 'data' when I'm reading the code.

File: test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl [hunk 3]
```diff
@@ -1,73 +1,69 @@
...
+    while readln(data_in) {
+      data += data_in;
 
       // Look for the start of a section, and if possible 
       // spawn a task to start work on the previous section.
-      if data[idx] == greaterThan {
-        if start == 0 then start = idx;
-        else {
+      if data.buff[idx] == greaterThan {
```
Comment (benharsh) [inline test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl]: We can probably use regular string comparison here as well.

File: test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl (hunk 1 above)
Comment (psahabu) [inline test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl]: I could go either way. We're already doing .buff operations in this section so it's not out of place, but it does look a bit odd.

File: test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl (hunk 3 above)
Comment (psahabu) [inline test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl]: While I like this idea, we'll need to convert between buff and string indices, either for this line or between main() and process(). I don't think there's a way to do it without making the code look funny.

File: test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl (hunk 1 above)
Comment (psahabu) [inline test/studies/shootout/reverse-complement/bharshbarg/revcomp-begin.chpl]: Looking at it again, we'd have to do the same awkward index conversion here too.

**Comment by psahabu** [comment]
The reference version for this implementation of revcomp is not very strong. @benharsh has found a better one.

**Pull request closed without merging**